File: FitTrackAPI/Models/Membership.cs
using System.ComponentModel.DataAnnotations.Schema;

public class Membership
{
    public int Id { get; set; }
    [ForeignKey("User")]
    public int UserId { get; set; }                
    public int GymId { get; set; }                 
    public int SessionsRemind { get; set; }       
    public DateTime ExpirationDate { get; set; }   
    public decimal Cost { get; set; }      

    public User User { get; set; }           
    public Gym Gym { get; set; }
}

File: fittrack_mobile_app/lib/models/membership.dart
import 'user.dart';
import 'gym.dart';

class Membership {
  int id;
  int userId;
  int gymId;
  int sessionsRemind;
  DateTime expirationDate;
  double cost;
  User user;
  Gym gym;

  Membership({
    required this.id,
    required this.userId,
    required this.gymId,
    required this.sessionsRemind,
    required this.expirationDate,
    required this.cost,
    required this.user,
    required this.gym,
  });

  factory Membership.fromJson(Map<String, dynamic> json) {
    return Membership(
      id: json['id'],
      userId: json['userId'],
      gymId: json['gymId'],
      sessionsRemind: json['sessionsRemind'],
      expirationDate: DateTime.parse(json['expirationDate']),
      cost: json['cost'].toDouble(),
      user: User.fromJson(json['user']),
      gym: Gym.fromJson(json['gym']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'userId': userId,
      'gymId': gymId,
      'sessionsRemind': sessionsRemind,
      'expirationDate': expirationDate.toIso8601String(),
      'cost': cost,
      'user': user.toJson(),
      'gym': gym.toJson(),
    };
  }
}
